StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1414InternalFieldMustBeAutoProperty.cs

```diff
@@ -86,7 +86,10 @@ private static bool IsInternalConst(IFieldSymbol fieldDeclarationSyntax)
 
             private static bool IsProtectedInternal(IFieldSymbol fieldDeclarationSyntax)
             {
-                return fieldDeclarationSyntax.DeclaredAccessibility == Accessibility.ProtectedAndInternal;
+                // Seems Microsoft.CodeAnalysis.CSharp has a issue with this being "ProtectedAndInternal"
+                // and detects them as "ProtectedOrInternal". Not sure if by design or just flat out a
+                // or because the version used is not the newest version bug.
+                return fieldDeclarationSyntax.DeclaredAccessibility == Accessibility.ProtectedOrInternal;
```

Comment:
📝 Here's the mapping:

| C# modifier | `Accessibility` |
| --- | --- |
| `protected internal` | `Accessibility.ProtectedOrInternal` |
| `private protected` | `Accessibility.ProtectedAndInternal` |

Reply:
oh, I see.